src/dmclock_server.h

```diff
@@ -204,17 +219,16 @@ namespace crimson {
     private:
 
       static double tag_calc(const Time time,
-			     double prev,
-			     double increment,
-			     uint32_t dist_req_val,
-			     bool extreme_is_high) {
+			     const double prev,
+			     const double increment,
+			     const uint32_t dist_req_val,
+			     const bool extreme_is_high,
+			     const Cost cost) {
 	if (0.0 == increment) {
 	  return extreme_is_high ? max_tag : min_tag;
 	} else {
-	  if (0 != dist_req_val) {
-	    increment *= dist_req_val;
-	  }
-	  return std::max(time, prev + increment);
+	  double tag_increment = increment * (dist_req_val + cost);
```

Comment:
@cbodley This is the key calculation with _cost_. We're adding two `uint32_t`'s and then multiplying by a `double`. If we're concerned about overflow, we could cast one of the addends to a uint64_t to force `uint64_t` addition, as in:
`double tag_increment = increment  * (uint64_t(dist_req_val) + cost);`